test/integration/smoke/test_activate_template.py

```diff
@@ -0,0 +1,196 @@
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from marvin.lib.base import (Account,
+                             ServiceOffering,
+                             VirtualMachine,
+                             DiskOffering,
+                             Template,
+                             Volume,
+                             Zone)
+from marvin.lib.common import (get_domain,
+                               get_zone,
+                               get_template)
+from marvin.cloudstackTestCase import cloudstackTestCase, unittest
+from nose.plugins.attrib import attr
+import time
+from marvin.cloudstackAPI import (registerTemplate, listOsTypes, activateSystemVMTemplate, createZone, listConfigurations, getSystemVMTemplateDefaultUrl)
+from marvin.lib.utils import cleanup_resources
+
+class TestActivateTemplate(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.testClient = super(TestActivateTemplate, cls).getClsTestClient()
+        cls.apiclient = cls.testClient.getApiClient()
+        cls.dbclient = cls.testClient.getDbConnection()
+        cls._cleanup = []
+        cls.templates = []
+
+        cls.services = cls.testClient.getParsedTestDataConfig()
+        cls.unsupportedHypervisor = False
+        cls.hypervisor = cls.testClient.getHypervisorInfo()
+        if cls.hypervisor.lower() not in ['kvm']:
+            # Only testing kvm
+            cls.unsupportedHypervisor = True
+            return
+
+        # Get Zone, Domain and templates
+        cls.domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
+        cls.services["mode"] = cls.zone.networktype
+        cls.services["virtual_machine"]["zoneid"] = cls.zone.id
+        cls.account = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            admin=True,
+            domainid=cls.domain.id
+        )
+        cls._cleanup.append(cls.account)
+        cls.user = Account.create(
+            cls.apiclient,
+            cls.services["account"],
+            domainid=cls.domain.id
+        )
+        cls._cleanup.append(cls.user)
+        cls.service_offering = ServiceOffering.create(
+            cls.apiclient,
+            cls.services["service_offerings"]["tiny"]
+        )
+        cls._cleanup.append(cls.service_offering)
+    
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cleanup_resources(cls.apiclient, cls._cleanup)
+        except Exception as e:
+            raise Exception("Warning: Exception during cleanup : %s" % e)
+        return
+
+    def setUp(self):
+        # Official cloudstack system vm template
```

Comment:
Assuming just one hypervisor is not a good idea, this would run OK on all hypervisors, since we're not really using the template, but if we were to use this one with vmware it'll fail. Just keep in mind, if you're writing test just for KVM, in setup check if hypervisor is KVM, else skip the test, which is not this case

Reply:
Done